bug: Properties ionNavWillChange and ionNavDidChange do not exist on IonNav

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

When upgrading to latest Ionic (v7.7.3) the build fails with error:
```
 error TS2339: Property 'ionNavDidChange' does not exist on type 'IonNav'.

44     this.ionNav.ionNavDidChange.subscribe(async () => {
                   ~~~~~~~~~~~~~~~
```

### Expected Behavior

This was not added to the breaking changes, and is still referenced in the documentation.
so the code should still work

### Steps to Reproduce

1. create a property of type `IonNav`
2. try to access ionNavDidChange
3. (IDE will probably error here)
4. run build

### Code Reproduction URL

https://stackblitz.com/edit/ionic-ion-nav-issue-reproduction?file=src%2Fmain.ts

### Ionic Info

Ionic:

   Ionic CLI : 5.4.16

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v21.1.0
   npm    : 10.2.5
   OS     : macOS Unknown


### Additional Information

Bug was already mentioned in https://github.com/ionic-team/ionic-framework/issues/28863 but closed without checking it properly.

--- a/packages/angular/common/src/directives/navigation/nav.ts
+++ b/packages/angular/common/src/directives/navigation/nav.ts
@@ -1,4 +1,4 @@
-import { ElementRef, Injector, EnvironmentInjector, NgZone, ChangeDetectorRef, Directive } from '@angular/core';
+import { ElementRef, Injector, EnvironmentInjector, NgZone, ChangeDetectorRef, Directive, EventEmitter } from '@angular/core';
 import type { Components } from '@ionic/core';
 
 import { AngularDelegate } from '../../providers/angular-delegate';
@@ -23,7 +23,16 @@ const NAV_METHODS = [
 ];
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
-export declare interface IonNav extends Components.IonNav {}
+export declare interface IonNav extends Components.IonNav {
+  /**
+   * Event fired when the nav will change components
+   */
+  ionNavWillChange: EventEmitter<CustomEvent<void>>;
+  /**
+   * Event fired when the nav has changed components
+   */
+  ionNavDidChange: EventEmitter<CustomEvent<void>>;
+}
 
 @ProxyCmp({
   inputs: NAV_INPUTS,

--- a/packages/angular/common/src/directives/navigation/nav.ts
+++ b/packages/angular/common/src/directives/navigation/nav.ts
@@ -1,4 +1,12 @@
-import { ElementRef, Injector, EnvironmentInjector, NgZone, ChangeDetectorRef, Directive, EventEmitter } from '@angular/core';
+import {
+  ElementRef,
+  Injector,
+  EnvironmentInjector,
+  NgZone,
+  ChangeDetectorRef,
+  Directive,
+  EventEmitter,
+} from '@angular/core';
 import type { Components } from '@ionic/core';
 
 import { AngularDelegate } from '../../providers/angular-delegate';

--- a/packages/angular/common/src/directives/navigation/nav.ts
+++ b/packages/angular/common/src/directives/navigation/nav.ts
@@ -30,7 +30,6 @@ const NAV_METHODS = [
   'getPrevious',
 ];
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
 export declare interface IonNav extends Components.IonNav {
   /**
    * Event fired when the nav will change components

